imap/smtpclient.c

```diff
@@ -143,6 +146,31 @@ static void smtpclient_logerror(smtpclient_t *sm, const char *cmd, int r);
 
 EXPORTED int smtpclient_open(smtpclient_t **smp)
 {
+    if (!smtpclient_ehlo_hostname) {
+        if (config_getswitch(IMAPOPT_CLIENT_BIND)) {
+            smtpclient_ehlo_hostname =
+                config_getstring(IMAPOPT_CLIENT_BIND_NAME);
+        }
+        if (!smtpclient_ehlo_hostname) {
+            smtpclient_ehlo_hostname = config_getstring(IMAPOPT_SERVERNAME);
+        }
+        if (!smtpclient_ehlo_hostname) {
+            xsyslog(LOG_ERR,
+                    "can not determine EHLO hostname, "
+                    "either client_bind/client_bind_name or servername option must be set",
+                    NULL);
+            return IMAP_INTERNAL;
+        }
```

Comment:
don't use `config_getstring(IMAPOPT_SERVERNAME)`, use `config_servername`, which is always set to something because it falls back on `gethostname(2)` (as documented)

Reply:
I now created https://github.com/cyrusimap/cyrus-imapd/pull/4971